src/nucleus/nucleus.ml

```diff
@@ -193,11 +194,26 @@ let print_is_term_abstraction = Nucleus_print.is_term_abstraction
 let print_is_type_abstraction = Nucleus_print.is_type_abstraction
 let print_eq_term_abstraction = Nucleus_print.eq_term_abstraction
 let print_eq_type_abstraction = Nucleus_print.eq_type_abstraction
-let print_judgement_abstraction = Nucleus_print.judgement_abstraction
 let print_boundary_abstraction = Nucleus_print.boundary_abstraction
+let print_judgement_with_boundary_abstraction = Nucleus_print.judgement_with_boundary_abstraction
 let print_derivation = Nucleus_print.derivation
 let print_error = Nucleus_print.error
 
+let print_judgement_abstraction ?max_level ~penv ~sgn abstr ppf =
```

Comment:
This is the wrong place to put an implementation. It should be in `Nucleus_print`. In any case, it is not appropriate to require `~sgn` as an argument here: either `sgn` is part of `penv` or we do not need it. In fact, we should *not* put `sgn` inside `penv`. Instead, just reinstate the old `print_judgement_abstraction` to what it used to be.

If anyone wants to print a judgement with a boundary, but only has the judgement, they will do it in two steps: first compute the judgement with boundary from the judgement, then print it.